docs/publish.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+set -e
+
+DOCS_DIR=$1
+PUBLISH_DIR=$2
+BUILD_SHA=`git rev-parse HEAD`
+
+if [[ ! -d $PUBLISH_DIR ]]; then
+    echo "$PUBLISH_DIR does not exist. Clone a fresh envoy repo."
+fi
+
+git -C $PUBLISH_DIR fetch
+git -C $PUBLISH_DIR checkout -B gh-pages origin/gh-pages
+rm -fr $PUBLISH_DIR/*
+cp -r $DOCS_DIR/* $PUBLISH_DIR
+git -C $PUBLISH_DIR add .
+git -C $PUBLISH_DIR commit -m "docs @$BUILD_SHA"
+
+echo
+echo "*** YOU MUST MANUALLY PUSH the gh-pages branch after verifying the commit ***"
```

Comment:
do i need to do anything else after push to origin?

can you put in some docs that, `make publish_docs` should be enough or we need to do something else